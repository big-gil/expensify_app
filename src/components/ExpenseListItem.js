import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";

const ExpenseListItem = ({ id, description, amount, createDate }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>

    <p>
      {amount} - {moment(createDate).format("MM/DD/YYYY")}
    </p>
  </div>
);

export default ExpenseListItem;
