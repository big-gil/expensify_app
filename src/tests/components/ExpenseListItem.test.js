import ExpenseListItem from "../../components/ExpenseListItem";
import { shallow } from "enzyme";
import React from "react";
import { expenses } from "../fixtures/expenses";

test("Expense List Item Test 1 - List Expenses", () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[2]} />);
  expect(wrapper).toMatchSnapshot();
});
