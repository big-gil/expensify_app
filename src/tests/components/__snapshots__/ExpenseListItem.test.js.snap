// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense List Item Test 1 - List Expenses 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/123abc"
  >
    <h3 />
  </Link>
  <p>
    122000
     - 
    01/02/1970
  </p>
</div>
`;
